src/System.Private.ServiceModel/src/System/ServiceModel/Dispatcher/StreamFormatter.cs

```diff
@@ -94,7 +103,10 @@ internal class SerializeAsyncResult : AsyncResult
             {
                 _streamFormatter = streamFormatter;
                 _writer = writer;
-                throw ExceptionHelper.PlatformNotSupported("StreamFormatter.SerializeAsyncResult is not supported.");
+
+                // As we use the Task-returning method for async operation,
+                // we shouldn't get to this point. Throw exception just in case.
+                throw new NotImplementedException();
```

Comment:
`throw ExceptionHelper.AsError(NotImplemented.ByDesign);` if we'll never get here. Consider getting rid of the rest of the implementation if the method shouldn't be implemented at all.